### DISCUSSION TITLE
Resolving State | Ligo Protocol

### DISCUSSION THREAD

**Discussion by giscus** [Ligo-Protocol/ligo-book#3] [Comments]
# Resolving State | Ligo Protocol

During a dispute, a third party is needed to independently resolve the latest state. This third party is a Chainlink External Adapter.

https://ligo-book.codyhatfield.me/Transacting/Enforcing%20Agreements/Resolving%20State

**Comment by codynhat** [comment]
@madhukar123456 Does every Chainlink External Adapter have a smart contract associated with it? Is this what the `Oracle.sol` is?

**Reply by codeicey** [reply]
External Adapter is one of the tools necessary for a smart contract that wants to get data from API. A smart contract can call API through chainlink node's job that gets data from the external adapter. 

The use of Oracle.sol is to set fullfillment request for the chainlink node address. Then the oracle.sol's address is used in job request in the chainlink node. 

Similary, for multiple responses and larger data, Operator.sol is used instead in the same way. 

In a nutshell, Chainlink Node has a direct need for Oracle.sol. Fetching data from a smart contract needs chainlink node with a job request to an External Adapter so Oracle.sol is like an indirect use and necessary step.

Yes, it has an associated smart contract like Oracle.sol or Operator.sol.

Docs: https://docs.chain.link/docs/fulfilling-requests/#:~:text=Chainlink%20nodes%20can%20fulfill%20requests,tasks%20to%20fulfill%20the%20request.
